apps/src/code-studio/components/progress/SectionSelector.jsx

```diff
@@ -22,18 +29,37 @@ const SectionSelector = React.createClass({
   },
 
   handleSelectChange(event) {
-    this.props.selectSection(event.target.value);
+    const newSectionId = event.target.value;
+    this.props.selectSection(newSectionId);
+    const newQueryString = queryString.stringify({
+      ...queryString.parse(location.search),
+      section_id: newSectionId === NO_SECTION ? undefined : newSectionId
+    });
+
+    // Push section_id update to query string (but don't do a navigation)
+    let newLocation = window.location.pathname;
+    if (newQueryString) {
+      newLocation += '?' + newQueryString;
+    }
+    window.history.pushState(null, document.title, newLocation);
```

Comment:
Not 100% sure this is the best place for this. Basic idea is that whenever we change the selected section, we want to change the query param in the URL (which will be persisted as well follow links). However, only when on a lesson page do we want to also do a page load (which we accomplish via this.props.onChange).